doc/sphinx/user-extensions/syntax-extensions.rst

```diff
@@ -29,30 +29,63 @@ Notations
 Basic notations
 ~~~~~~~~~~~~~~~
 
-.. cmd:: Notation
+.. cmd:: Notation @string := @one_term {? ( {+, @syntax_modifier } ) } {? : @ident }
 
-   A *notation* is a symbolic expression denoting some term or term
-   pattern.
+   .. insertprodn syntax_modifier syntax_extension_type
 
-A typical notation is the use of the infix symbol ``/\`` to denote the
-logical conjunction (and). Such a notation is declared by
+   .. prodn::
+      syntax_modifier ::= at level @num
+      | in custom @ident
+      | in custom @ident at level @num
+      | left associativity
+      | right associativity
+      | no associativity
+      | only printing
+      | only parsing
+      | format @string {? @string }
+      | @ident , {+, @ident } at @level
+      | @ident at @level {? @constr_as_binder_kind }
+      | @ident @constr_as_binder_kind
+      | @ident @syntax_extension_type
+      level ::= level @num
+      | next level
+      constr_as_binder_kind ::= as ident
```

Comment:
Was called `binderinterp` in the former `productionlist`. Probably worth keeping that name.